plugins/gevent/gevent.c

```diff
@@ -389,7 +389,7 @@ static void gil_gevent_get() {
 }
 
 static void gil_gevent_release() {
-	PyGILState_Release((PyGILState_STATE) pthread_getspecific(up.upt_gil_key));
+	PyGILState_Release((long) pthread_getspecific(up.upt_gil_key));
```

Comment:
Please keep the proper type `(PyGILState_STATE)(long)`

Reply:
Nice! I'm not a C expert so I wouldn't know.

I have looked into the code and the function `pthread_getspecific` has different casts throughout all the code. Is that intentional? If not, it's just as a security risk as this one, imo.

Reply:
It returns void * so you are supposed to cast it to whatever you set it before to.